products/servicedirectory/terraform.yaml

```diff
@@ -0,0 +1,59 @@
+# Copyright 2020 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Provider::Terraform::Config
+overrides: !ruby/object:Overrides::ResourceOverrides
+  Namespace: !ruby/object:Overrides::Terraform::ResourceOverride
+    import_format: ["projects/{{project}}/locations/{{location}}/namespaces/{{namespace_id}}"]
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "service_directory_namespace_basic"
+        primary_resource_id: "example"
+        vars:
+          namespace_id: "example-namespace"
+        min_version: beta
+    properties:
+      location: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true
+  Service: !ruby/object:Overrides::Terraform::ResourceOverride
+    import_format: ["{{namespace}}/services/{{service_id}}"]
+    examples:
+      - !ruby/object:Provider::Terraform::Examples
+        name: "service_directory_service_basic"
+        primary_resource_id: "example"
+        vars:
+          service_id: "example-service"
+          namespace_id: "example-namespace"
+        min_version: beta
+    properties:
+      namespace: !ruby/object:Overrides::Terraform::PropertyOverride
+        ignore_read: true
+  Endpoint: !ruby/object:Overrides::Terraform::ResourceOverride
+    import_format: ["{{service}}/endpoints/{{name}}"]
```

Comment:
You may end up needing a self_link_as_name (or similar) custom import, since the  `name` field will contain slashes. Beta-only resources don't generate import steps for reasons that I can't remember off the top of my head, but if you're going to do handwritten update tests anyway, those should confirm whether you need to adjust the import logic. You might also be able to get around it by using `endpoint_id` and the full link format instead of `name`